File: fast builder/app/src/main/java/fastbuilder/app/model/CharSheet.java
package fastbuilder.app.model;

import fastbuilder.app.model.background.Background;
import fastbuilder.app.model.race.Race;
import fastbuilder.app.model.subclass.CharClass;

public class CharSheet {
    Background background;
    Race race;
    CharClass charClass;
    public Background getBackground() {
        return background;
    }
    public void setBackground(Background background) {
        this.background = background;
    }
    public Race getRace() {
        return race;
    }
    public void setRace(Race race) {
        this.race = race;
    }
    public CharClass getSubclass() {
        return charClass;
    }
    public void setSubclass(CharClass charClass) {
        this.charClass = charClass;
    }
    
    public CharSheet() {
    }
    public CharSheet(Background background, Race race, CharClass charClass) {
        this.background = background;
        this.race = race;
        this.charClass = charClass;
    }

    

    
}

File: sbnz-2023/front/src/app/model/CharSheet.ts
export class CharSheet {
  background: string | undefined;
  race: string | undefined;
  charClass: string | undefined;
  subclass: string | undefined;
  proficiencies: string[] | undefined;
  abilityScores: { [key: string]: number } | undefined;
  classString: string | undefined;

  constructor() {
    this.background = undefined;
    this.race = undefined;
    this.charClass = undefined;
    this.subclass = undefined;
    this.proficiencies = undefined;
    this.abilityScores = undefined;
    this.classString = undefined;
  }

  getBackground() {
    return this.background === undefined ? '' : this.background
  }
  getRace() {
    return this.race === undefined ? '' : this.race
  }
  getCharClass() {
    return this.charClass === undefined ? '' : this.charClass
  }
  getSubClass() {
    return this.subclass === undefined ? '' : this.subclass
  }
  getClassString() {
    return this.classString === undefined ? '' : this.classString
  }
  getProficiencies() {
    return this.proficiencies === undefined ? [] : this.proficiencies
  }

  isReady(): boolean {
    return this.abilityScores !== undefined && this.race !== undefined && this.charClass !== undefined &&
      this.subclass !== undefined && this.proficiencies !== undefined
  }


}
